Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -312,35 +312,105 @@ PhysicalParticleContainer::AddGaussianBeam (
 }
 
 void
-PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f, amrex::Real q_tot)
+PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
+                                               amrex::Real physical_q_tot)
 {
 #ifdef WARPX_USE_OPENPMD
     openPMD::Series series = openPMD::Series(s_f, openPMD::AccessType::READ_ONLY);
     amrex::Print() << "openPMD standard version " << series.openPMD() << "\n";
-
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(series.iterations.size() == 1u, "External "
                                      "file should contain only one iteration\n");
     openPMD::Iteration& i = series.iterations[1];
-
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(i.particles.size() == 1u, "External file "
                                      "should contain only one species\n");
     std::pair<std::string,openPMD::ParticleSpecies> ps = *i.particles.begin();
 
-    //TODO: In future PRs will add AMREX_ALWAYS_ASSERT_WITH_MESSAGE to test if mass and charge are both const
+    //TODO: In future PRs will add ASSERT_WITH_MESSAGE to test if mass and charge are both const
     amrex::Real p_m = ps.second["mass"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
+    double mass_unit = ps.second["mass"][openPMD::RecordComponent::SCALAR].unitSI();
     amrex::Real p_q = ps.second["charge"][openPMD::RecordComponent::SCALAR].loadChunk<amrex::Real>().get()[0];
-    int npart = ps.second["position"]["x"].getExtent()[0];
+    double charge_unit = ps.second["charge"][openPMD::RecordComponent::SCALAR].unitSI();
+    auto npart = ps.second["position"]["x"].getExtent()[0];
     series.flush();
+    
+    mass=mass*mass_unit;
+    charge=charge*charge_unit;
 
-    mass = p_m*PhysConst::mevpc2_kg;
-    charge = p_q*PhysConst::q_e;
-    Real const weight = q_tot/(charge*amrex::Real(npart));
+    amrex::Real weight;
+    if (physical_q_tot!=0.0){
+        weight = physical_q_tot/(p_q*charge_unit*amrex::Real(npart));
+    }
+    else {
+        weight = charge;
```

Comment:
I think not, if no conversion then this is reading from (for example) previous macro particle information - which is already with the correct weight - be it what it may be (i.e. !=0). Do you agree?

Reply:
Yes, agreed. Either one or the other.